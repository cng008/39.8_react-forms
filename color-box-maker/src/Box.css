.Box {
  color: white;
  font-size: 2em;
  display: block;
  text-align: center;
}
