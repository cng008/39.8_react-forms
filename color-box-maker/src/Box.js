import React from 'react'

/** display a div with a background color, width and height based on the props passed to it
 *
 *  * This has no state --- just three props
 *
 */

const Box = ({ id, backgroundColor, width, height }) => {
  const styles = {
    height: `${height}em`,
    width: `${width}em`,
    backgroundColor
  }
  return (
    <>
      <div className="Box" style={styles}></div>
    </>
  )
}
export default Box
