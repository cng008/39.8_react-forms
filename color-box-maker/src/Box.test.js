import React from 'react'
import { render } from '@testing-library/react'
import Box from './Box'

describe('<Box /> rendering', () => {
  it('renders without crashing', function() {
    render(<Box />)
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<Box />)
    expect(asFragment()).toMatchSnapshot()
  })
})
