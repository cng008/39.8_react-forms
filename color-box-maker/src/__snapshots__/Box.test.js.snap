// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> rendering matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <span>
      x
    </span>
  </div>
</DocumentFragment>
`;
